src/fetching/getBreadcrumbs.ts

```diff
@@ -0,0 +1,11 @@
+import { Breadcrumb } from '../types';
+import { defaultFetch } from '.';
+
+export default function getBreadcrumbs(pathname: string, queryString: string, fetch  = defaultFetch) {
+    return async () : Promise<Breadcrumb[]> => {
+        const bRes = await fetch(`/breadcrumbs?pathname=${pathname}&${queryString}`);
+        const bData = await bRes.json();
+        return bData.breadcrumbs;
```

Comment:
In this handler, we should throw an error if there's a non 200 status. Then, in the file `BreadcrumbTrail.tsx` we could conditionally show the trail only if react query has `query.isSuccess === true`. The way it works now works but it isn't so clear about errors, so this is just a code improvement. Not required. I think it's great you worked on improving this code anyway!

Reply:
I agree it looks like there should be error handling - it looks like the `/fetching` folder in general could do with attention in this respect! Maybe all fetch-related error handling could be incorporated into a separate issue? :-)

Reply:
Yeah, changing all "fetching" modules for consistency is on our radar along with a bigger refactor of queries, but we also try to make sure any new ones we create are correctly compatible with `react-query` (e.g. throw an error if they fail). So it would be great if you could make this one compatible as well, i.e. add these three lines before parsing the response:

```ts
if (!bRes.ok) {
  throw new APIError('GET', url);
}
```

Reply:
Will do!